github-push/handler.go

```diff
@@ -200,28 +212,28 @@ func readBool(key string) bool {
 	return false
 }
 
+// enableStatusReporting check if status reportng is enabled
 func enableStatusReporting() bool {
 	return os.Getenv("report_status") == "true"
 }
 
-func reportStatus(status *sdk.Status) {
-
-	if !enableStatusReporting() {
-		return
-	}
-
-	hmacKey, keyErr := sdk.ReadSecret("payload-secret")
-	if keyErr != nil {
-		log.Printf("failed to load hmac key for status, error " + keyErr.Error())
-		return
-	}
-
-	gatewayURL := os.Getenv("gateway_url")
+// initStatusReporter initialize status reporter
+func initStatusReporter() sdk.StatusReporter {
+	var reporter sdk.StatusReporter
+	if enableStatusReporting() {
+		hmacKey, keyErr := secret.Read("payload-secret")
+		if keyErr != nil {
+			log.Printf("failed to load hmac key for status, error " + keyErr.Error())
+			reporter = sdk.NilStatusReporter{}
+			return reporter
+		}
 
-	_, reportErr := status.Report(gatewayURL, hmacKey)
-	if reportErr != nil {
-		log.Printf("failed to report status, error: %s", reportErr.Error())
+		gatewayURL := os.Getenv("gateway_url")
+		reporter = sdk.GithubStatusReporter{HmacKey: hmacKey, Gateway: gatewayURL}
+	} else {
+		reporter = sdk.NilStatusReporter{}
 	}
+	return reporter
 }
```

Comment:
I think that instead of defining `var reporter sdk.StatusReporter` at the beginning of a function we could just return reporters. So for example:
instead of:
```
			reporter = sdk.NilStatusReporter{}
			return reporter
```
just:
```
			return sdk.NilStatusReporter{}
```
instead of:
```
		reporter = sdk.GithubStatusReporter{HmacKey: hmacKey, Gateway: gatewayURL}
```
this:
```
		return sdk.GithubStatusReporter{HmacKey: hmacKey, Gateway: gatewayURL}
```
then we can completely return last `else` statement from line 233 and do just:
```
return sdk.NilStatusReporter{}
```
what do you think about that @s8sg ?

Reply:
Apologies for not selecting all lines which this comment refer too